src/htcp.cc

```diff
@@ -282,8 +282,6 @@ htcpSyncAle(AccessLogEntryPointer &al, const Ip::Address &caddr, const int opcod
     al->setVirginUrlForMissingRequest(al->url);
     // HTCP transactions do not wait
     al->cache.start_time = current_time;
```

Comment:
I am concerned that in this and similar contexts, we use current_time for one half of the formula (e.g., response time end-start calculation) and RecordTime::systemTime for the other half. Yes, the two clocks or "now" values are closely related, but they are _updated_ at different times. Thus, when system time changes, we may observe odd cases where systemTime becomes smaller than current_time. I do not want us to introduce those uncertainties.

N.B. Ideally, various start/end times that are used for response time calculation should not be recorded in terms of current_time/system_clock. They should use steady_clock. However, fixing that is outside this work scope. Here, just want to avoid adding new sources of mismatching time values...

I believe we should add RecordTime::legacySystemTime that will keep a copy of current_time and use that legacySystemTime for calculations that used current_time before this PR (and that need to be adjusted using RecordTime approach, of course).

RecordTime::systemTime is for use with MessageTimer and its std::chrono::system_clock only.

RecordTime::stopwatchTime is for use with Stopwatch-based code and its std::chrono::steady_clock. We will probably rename this data member to RecordTime::steadyTime (and RecordTime to something like Moment), but that can be done later.

Reply:
Ok, I added RecordTime::legacySystemTime and RecordTime::systemTime became unused then. Should we keep it?